File: 2212-removing-minimum-and-maximum-from-array/removing-minimum-and-maximum-from-array.cpp
class Solution {
public:
    int minimumDeletions(vector<int>& nums) {
        int n = nums.size();
        int minInd = 0;
        int maxInd = 0;
        int mini = nums[0];
        int maxi = nums[0];
        for(int i=1; i<n; i++){
            if(nums[i]<mini){
                mini = nums[i];
                minInd = i;
            }
            if(nums[i]>maxi){
                maxi = nums[i];
                maxInd = i;
            }
        }
        int ans = min({max(minInd, maxInd)+1, n-min(minInd, maxInd), minInd+1+n-maxInd, n-minInd+maxInd+1});
        return ans;
    }
};

File: 2212-removing-minimum-and-maximum-from-array/removing-minimum-and-maximum-from-array.java
class Solution {
    public int minimumDeletions(int[] nums) {
        int n = nums.length;
        int minInd = 0, maxInd = 0;
        int mini = nums[0], maxi = nums[0];
        for (int i = 1; i < n; i++) {
            if (nums[i] < mini) { mini = nums[i]; minInd = i; }
            if (nums[i] > maxi) { maxi = nums[i]; maxInd = i; }
        }
        int i = Math.min(minInd, maxInd);
        int j = Math.max(minInd, maxInd);
        return Math.min(j + 1, Math.min(n - i, (i + 1) + (n - j)));
    }
}